changesLog.js

```diff
@@ -1,34 +1,75 @@
+const octokit = require('./githubAPI/octokit');
 const commander = require('commander');
-const { getIssuesByMilestone } = require('./changesLogGenerator');
+const fs = require('fs');
+const { fetchIssuesInMilestone, generateChangesLog } = require('./helpers');
 
-function commaSeparatedList(value, dummyPrevious) {
+function commaSeparatedList(value) {
   return value.split(',');
 }
 
+function checkValidStates(value) {
+  return (value === 'closed') || (value === 'open') || (value === 'all') ? value : undefined;
+}
+
 commander
   .version('1.0.0', '-v, --version')
   .usage('[OPTIONS]...')
-  .requiredOption('-m, --milestone <name>', 'must filter issue from milestone')
-  .option('-d, --duplicate <allow>', 'Should allow duplicated issues (Default: true)')
-  .option('-f, --filters <items>', 'comma separeted labels to filter issues', commaSeparatedList)
-  .option('-g, --groups <items>', 'comma separated labels to group by issues', commaSeparatedList)
+  .requiredOption('-m, --milestone <number>', 'must filter issue from milestone')
+  .option('-c, --customer <customer>, will filter the issues related to a customer if this option is received')
+  .option('-i, --includeAll <flag>', 'will include a section with issues done for all clients (Default: false)')
+  .option('-s, --state <state>', 'should use issues with state: closed, open, all (Default: closed)', checkValidStates)
+  .option('-f, --filters <labels>', 'comma separated labels to filter and group by issues', commaSeparatedList)
+  .option('-d, --duplicate <allow>', 'should allow duplicated issues in different groups (Default: true)')
   .parse(process.argv);
 
+let groupedIssues = [];
+const filters = commander.filters;
 const params = {
-  duplicate: commander.duplicate ? false : true,
   milestone: commander.milestone,
-  filters: commander.filters,
-  groups: commander.groups
+  state: commander.state ? commander.state : 'closed',
+  customer: commander.customer ? commander.customer : undefined,
+  duplicateIssues: commander.duplicate ? commander.duplicate : true,
+  includeIssueForAll: commander.includeAll ? commander.includeAll : true
```

Comment:
I think a bit nicer to use destructuring with defaults here:

```suggestion
  const { milestone, state = 'closed', customer = null, duplicate = true, includeAll: includeIssueForAll = true } = commander;
  const params = { milestone, state, customer, duplicate, includeIssueForAll }
```

Thoughts?

Reply:
Wow! That looks great. Thanks Will! I just didn't use because I had no idea it could be done that way. I'll add and test if will work the same.